contracts/WalletSimple.sol

```diff
@@ -430,26 +430,28 @@ contract WalletSimple {
   function tryInsertSequenceId(uint256 sequenceId) private onlySigner {
     // Keep a pointer to the lowest value element in the window
     uint256 lowestValueIndex = 0;
+    // fetch recentSequenceIds into memory for function context to avoid unnecessary sloads
+    uint256[10] memory _recentSequenceIds = recentSequenceIds;
     for (uint256 i = 0; i < SEQUENCE_ID_WINDOW_SIZE; i++) {
-      require(recentSequenceIds[i] != sequenceId, 'Sequence ID already used');
+      require(_recentSequenceIds[i] != sequenceId, 'Sequence ID already used');
 
-      if (recentSequenceIds[i] < recentSequenceIds[lowestValueIndex]) {
+      if (_recentSequenceIds[i] < _recentSequenceIds[lowestValueIndex]) {
         lowestValueIndex = i;
       }
     }
 
     // The sequence ID being used is lower than the lowest value in the window
     // so we cannot accept it as it may have been used before
     require(
-      sequenceId > recentSequenceIds[lowestValueIndex],
+      sequenceId > _recentSequenceIds[lowestValueIndex],
       'Sequence ID below window'
```

Comment:
lgtm. This makes me think maybe we should change access to the other variables we have too like signers.
And maybe  MAX_SEQUENCE_ID_INCREASE and SEQUENCE_ID_WINDOW_SIZE should be coded in instead of being defined in storage